test/e2e/serial/config/config.go

```diff
@@ -17,15 +17,27 @@
 package config
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
 	"os"
 
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+
 	. "github.com/onsi/gomega"
+
+	nrtv1alpha2 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha2"
+
+	"github.com/openshift-kni/numaresources-operator/pkg/validator"
+	e2eclient "github.com/openshift-kni/numaresources-operator/test/utils/clients"
+	e2enrt "github.com/openshift-kni/numaresources-operator/test/utils/noderesourcetopologies"
 )
 
 const (
-	MultiNUMALabel    = "numa.hardware.openshift-kni.io/cell-count"
-	nropTestCIImage   = "quay.io/openshift-kni/resource-topology-exporter:test-ci"
-	SchedulerTestName = "test-topology-scheduler"
+	MultiNUMALabel                   = "numa.hardware.openshift-kni.io/cell-count"
+	nropTestCIImage                  = "quay.io/openshift-kni/resource-topology-exporter:test-ci"
+	SchedulerTestName                = "test-topology-scheduler"
+	minNumberOfNodesWithSameTopology = 2
```

Comment:
we missed using this. If not mistaken, currently this code checks that the nrts are aligned with the kubeletconfig which is good and is needed. the suite fails if these configurations are not aligned, and that is the expected behavior. yet it is not enough to address the main problem. I expect the suite to still be executed with different topologies on different policies circling back to the main problem. 
I believe the only left thing to do is to check that there are at least 2 ( minNumberOfNodesWithSameTopology ) nodes with similar nrt.TopologyPolicy && it is equal to (either nrtv1alpha2.SingleNUMANodeContainerLevel or nrtv1alpha2.SingleNUMANodePodLevel). Let me know what you think.

Reply:
Thanks for all the modifications!
/lgtm
/hold 
I guess more improvements can be done in a later PR but hold for other reviewers' approval.